extensions/sql-database-projects/src/common/utils.ts

```diff
@@ -489,6 +490,23 @@ export async function retry<T>(
 	return undefined;
 }
 
+/**
+ * Detects whether the specified command-line command is available on the current machine
+ */
+export async function detectCommandInstallation(command: string): Promise<boolean> {
+	try {
+		const found = await which(command);
+
+		if (found) {
+			return true;
+		}
+	} catch (err) {
+		console.log(getErrorMessage(err));
+	}
+
+	return false;
+}
```

Comment:
Sadly only the sync version has an option to return undefined instead of throwing, hence why this is a helper method instead of one-liner.